dask/delayed.py

```diff
@@ -470,21 +470,31 @@ class Delayed(DaskMethodsMixin, OperatorMethodMixin):
     Equivalent to the output from a single key in a dask graph.
     """
 
-    __slots__ = ("_key", "dask", "_length")
+    __slots__ = ("_key", "dask", "_length", "_dask_layers")
 
     def __init__(self, key, dsk, length=None):
         self._key = key
         self.dask = dsk
         self._length = length
 
+        # `__dask_layers__` of an one-layered HLG is the layer name
+        # TODO: what is the name of a multi-layered HLG?
+        if isinstance(dsk, HighLevelGraph) and len(dsk.layers) == 1:
+            self._dask_layers = set(iter(dsk.layers))
+        else:
+            self._dask_layers = (self.key,)
+
     def __dask_graph__(self):
         return self.dask
 
     def __dask_keys__(self):
         return [self.key]
 
     def __dask_layers__(self):
-        return (self.key,)
+        if hasattr(self, "_dask_layers"):
+            return self._dask_layers
+        else:
+            return (self.key,)
```

Comment:
I would expect this to always be `self.key`.  I'm curious in which cases this assumption is incorrect.

Reply:
This boils down to the discrepancy between the definition of `__dask_layers__` that should return _output layers_ and `Delayed` that might not have any layers. 
`Delayed.key` is always a _low level_ graph key and so this is kind of a hack that make `Delayed` return a layer if possible.